cpp/src/arrow/python/datetime.h

```diff
@@ -35,6 +37,8 @@
 #define PyDateTimeAPI ::arrow::py::internal::datetime_api
 
 namespace arrow {
+using internal::MultiplyWithOverflow;
+using internal::AddWithOverflow;
```

Comment:
Either you can leave this out, or either you don't have to include the `arrow::internal::` below when using those, I think